src/core/adios_internals.c

```diff
@@ -385,7 +385,15 @@ int adios_parse_dimension (const char * dimension
     }
     else
     {
-        dim->dimension.rank = strtol (dimension, NULL, 10);
+        int errno_save = errno;
+        long t = strtol (dimension, &end, 10);
```

Comment:
@pnorbert The [C standard not guarantied](http://stackoverflow.com/a/18901178) that `long` is 64bit. The size need at least the size of `int`. The save way is to use `uin64_t` or `long long`